### PR TITLE
Bitwise methods

### PR DESCRIPTION
**Pull request by Vap0r1ze**
Hey there Mike, I've added some bitwise methods. I don't think I did the docs correctly, thanks!

### CODE REVIEW

**Comment by MikeMcl** [comment]
These look like neat implementations. 

A few issues:

1. Is it right to add bitwise methods to the library given that they only operate on the integer part of a number?

2. Wouldn't these be better as an extension to the library that can be added to `BigNumber.prototype` as required?

3. I am committed to having the library run on JavaScript engines that only implement ECMAScript 3, so the use of `String.prototype.repeat` is unacceptable. (Replace by a `repeat` function?)

4. 
```js
x = new BigNumber(2);
x.shiftRight(2);    // 'Error: new BigNumber() not a base 2 number: '
```

5. There is already a [shift](http://mikemcl.github.io/bignumber.js/#shift) method, though it is currently missing from the API image in the README. (Perhaps use `bitShiftLeft` etc. instead?)

6. [Tests](https://github.com/MikeMcl/bignumber.js/tree/master/test) and [documentation](http://mikemcl.github.io/bignumber.js/) are required. 

7. Minor code issues such as calling `new BigNumber(a)` several times in a method rather than just once and assigning the result to a variable. The same with `.length`. `n = +n.toString()` is redundant if you are using `-n`  or `+n`. For consistency, I'd prefer it if the `isValidInt` function was used to check the argument in the same way as `toFixed` etc. - though I appreciate that it's usage can be a bit hard to follow.

Basically, you have done the fun part of writing the methods, but there is quite a bit more work necessary before I could accept the PR. I don't have time to do this myself at the moment, though I may have in future weeks/months.

PR under consideration. No decision reached yet.

**Comment by Vap0r1ze** [comment]
I should probably make this an addon.
